infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LRRollingUpgradeHandler.java

```diff
@@ -74,7 +75,7 @@ public LRRollingUpgradeHandler(ILogReplicationVersionAdapter versionAdapter, Cor
         }
     }
 
-    public boolean isLRUpgradeInProgress(TxnContext txnContext) {
+    public boolean isLRUpgradeInProgress(CorfuStore corfuStore, TxnContext txnContext) {
```

Comment:
can we get rid of the CorfuStore passed to the constructor, then?

We can also get rid of opening the event table on L67 now, right?

Reply:
Can we just do it the other way then ? Stash the `corfuStore` in the constructor and also open the tables ? that will keep the actual calling function free from extra threads?